app/Services/TaskUrlGenerator.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace App\Services;
+
+use App\Models\User;
+use App\Models\Customer;
+use App\Models\Supplier;
+
+class TaskUrlGenerator
+{
+    private const BASE_URL = 'https://example.com/tasks';
```

Comment:
and this  (BASE_URL) constance would not be needed if  the frontend has an env for base route, this base route would only be important for the task email, ok i see what you did there, so we can handle the base url from our env file on the backend, so the frontend does not have to handle it eg env("FRONTENDBASEURL") etc